### PR TITLE
docs: Cleanup docs readme

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Cleanup docs readme
```

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-  - /pkg/cloud/azure/actuators/cluster - cluster actuator for cluster api provider
-  - /pkg/cloud/azure/actuators/machine - machine actuator for cluster api provider
-  - /pkg/cloud/azure/services - each folder represents an azure service, implements Get/CreateOrUpdate/Delete
-  - /pkg/cloud/azure/services/config - contains startup scripts for controlplane and worker nodes
-
-## Components
-
-### Cluster Actuator
-
-The cluster actuator creates the following resources, in order:
```
Comment (CecileRobertMichon) [inline docs/design.md:18]: this is very outdated, removing it to avoid confusion until we can rewrite it

**Review by mboersma — APPROVED** [review]
lgtm

**Review by devigned — APPROVED** [review]
/lgtm
/approve

**Review by CecileRobertMichon — COMMENTED** [review]
/hold

**Comment by CecileRobertMichon** [comment]
/hold cancel

there was an extra space...

**Pull request merged**